Source/MOLAuthenticatingURLSession/MOLAuthenticatingURLSession.m

```diff
@@ -371,10 +371,10 @@ - (SecIdentityRef)identityByFilteringArray:(NSArray *)array
       return identityRef;
     } else {
       // Avoid infinite recursion from self-signed certs
-      if ((cert.commonName && [cert.commonName isEqual:cert.issuerCommonName]) &&
-          (cert.countryName && [cert.countryName isEqual:cert.issuerCountryName]) &&
-          (cert.orgName && [cert.orgName isEqual:cert.issuerOrgName]) &&
-          (cert.orgUnit && [cert.orgUnit isEqual:cert.issuerOrgUnit])) {
+      if ((!cert.commonName || [cert.commonName isEqual:cert.issuerCommonName]) &&
```

Comment:
I think the `isEqual:` usage is intentional here (vs. `isEqualToString:`), but maybe add a note for future readers.

Reply:
I can't remember what the reason was, if there is one, other than there not being any real advantage to `isEqualToString:`